src/hotspot/share/memory/heapShared.cpp

```diff
@@ -685,13 +681,13 @@ void HeapShared::resolve_classes(TRAPS) {
   }
   resolve_classes_for_subgraphs(closed_archive_subgraph_entry_fields,
                                 num_closed_archive_subgraph_entry_fields,
-                                CHECK);
+                                THREAD /* exceptions are ignored */);
   resolve_classes_for_subgraphs(open_archive_subgraph_entry_fields,
                                 num_open_archive_subgraph_entry_fields,
-                                CHECK);
+                                THREAD /* exceptions are ignored */);
   resolve_classes_for_subgraphs(fmg_open_archive_subgraph_entry_fields,
                                 num_fmg_open_archive_subgraph_entry_fields,
-                                CHECK);
+                                THREAD /* exceptions are ignored */);
```

Comment:
Are exceptions ignored or unexpected? This pattern doesn't exist anywhere else as far as I know, otherwise maybe we should have a CLEAR macro in addition to CHECK/CATCH that clears pending exceptions? I don't really like that there may be exceptions lurking through multiple calls.

Reply:
Ok, so I'm reading this in context. This still looks unusual and possibly problematic without deeper inspection.  It seems like removing TRAPS here also and the callees, and adding this before the functions that clear pending exceptions would be better.
JavaThread* THREAD = JavaThread::current();